src/domain/challenge/base-challenge/journey.interface.ts

```diff
@@ -0,0 +1,28 @@
+import { Field, InterfaceType } from '@nestjs/graphql';
+import { RelationshipNotFoundException } from '@common/exceptions';
+import { LogContext } from '@common/enums';
+import { IChallenge } from '../challenge/challenge.interface';
+import { IOpportunity } from '@domain/collaboration/opportunity';
+import { ISpace } from '../space/space.interface';
+import { ICollaboration } from '@domain/collaboration/collaboration';
+import { INameable } from '@domain/common/entity/nameable-entity';
+
+@InterfaceType('Journey', {
+  resolveType(journey) {
+    if (!journey.innovationFlow) return ISpace;
+    if (journey.parentSpace) return IChallenge;
+    if (journey.challenge) return IOpportunity;
+
+    throw new RelationshipNotFoundException(
+      `Unable to determine Journey type for ${journey.id}`,
+      LogContext.CHALLENGES
+    );
+  },
+})
+export abstract class IJourney extends INameable {
```

Comment:
isn't it better to extend `IBaseChallenge` instead - it's 1:1?

Reply:
/home/valentin/work/repos/alkemio/server/src/domain/challenge/challenge/challenge.interface.ts:9
export abstract class IChallenge extends IJourney {
                                         ^
TypeError: Class extends value undefined is not a constructor or null
    at Object.<anonymous> (/home/valentin/work/repos/alkemio/server/src/domain/challenge/challenge/challenge.interface.ts:9:42)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/valentin/work/repos/alkemio/server/src/domain/challenge/base-challenge/journey.interface.ts:4:1)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)

That's what I get in this scenario.